packages/aggregator/src/analytics/aggregateAnalytics/aggregations/sumPreviousPerPeriod.js

```diff
@@ -87,3 +94,13 @@ const calculatePeriodsFromAnalytics = (analytics, aggregationPeriod, requestedPe
 
   return getPeriodsInRange(startPeriod, endPeriod);
 };
+
+export const getDateRangeForSumPreviousPerPeriod = (dateRange, _, periodType) => {
```

Comment:
Using classes for aggregations could allow for a nice pattern, where two public methods would be part of their interface: `aggregate`, and `getDateRange`

Reply:
Yeah, it really seems like a good case from some class based logic here huh...

Still best to leave things as is for now.